PR471 review: various fixes (2)

dmidem opened the pull request:
Address the following review comments:

r2585521988
r2585536670
r2585544143
r2585546655
r2585548116
r2585631816
r2585636883
r2599516192
r2599562825
r2599565711
r2599584010
r2599589396
r2599594554
r2599594936
r2599745324
r2602373392
r2602378140
r2602410770
r2602415043
r2602416292
r2602421894
r2602426619
r2602532807
r2602541601
r2602556915
r2602570477 (see note)
r2602938883 (see note)
r2602418341

Note: r2602570477 and r2602938883 — we renamed the generic parameter P to Pr for OrchardPrimitives everywhere for consistency

dmidem committed:
Address the following review comments:

r2585521988
r2585536670
r2585544143
r2585546655
r2585548116
r2585631816
r2585636883
r2599516192
r2599562825
r2599565711
r2599584010
r2599589396
r2599594554
r2599594936
r2599745324
r2602373392
r2602378140
r2602410770
r2602415043
r2602416292
r2602418341
r2602421894
r2602426619
r2602532807
r2602541601
r2602556915
r2602570477 (see note
r2602938883 (see note)

Note: r2602570477 and r2602938883 — we renamed the generic parameter
P to Pr only where it conflicts with P for Proof. Should we use Pr for
OrchardPrimitives everywhere else for consistency?

ConstanceBeguier commented on src/builder.rs:
```diff
@@ -61,9 +61,9 @@ pub enum BundleType {
     /// The default bundle type has all flags enabled, ZSA disabled, and does not require a bundle
-    /// to be produced.
+    /// to be produced if no spends or outputs have been added to the bundle.
     pub const DEFAULT_VANILLA: BundleType = BundleType::Transactional {
```
See r2602373392
```suggestion
    pub const DEFAULT: BundleType = BundleType::Transactional {
```

dmidem replied:
Fixed (renamed `BundleType::DEFAULT_VANILLA` back to `DEFAULT`).

ConstanceBeguier commented on src/builder.rs:
```diff
@@ -537,11 +539,6 @@ impl ActionInfo {
     }
 
     fn build_for_pczt(self, mut rng: impl RngCore) -> crate::pczt::Action {
-        assert_eq!(
```
We should discuss this change at the ECC meeting.
Currently, when creating a pczt::Action, we do not verify that the spend asset and the output asset are the same. One option would be to introduce a new function to initialize a pczt::Action, where this check would be enforced.
For now, I suggest removing this change from the current PR and also removing r2602418341 from the PR description.

dmidem replied:
Fixed (removed `assert_eq` from the `ActionInfo` constructor and return it to `build` and `build_for_pczt` methods, also removed `r2602418341` from the PR description, changed `FIXED` to `TODO` in #201 comments)

ConstanceBeguier replied:
I reverted the commit “Rename P: OrchardPrimitives to Pr everywhere in the code”.
It is safe to remove from ActionInfo::build_for_pczt the check that the output asset and the spend asset are equal because this function already verifies that:
- the spend asset is equal to the native asset in self.spend.into_pczt(...),
- the output asset is equal to the native asset in self.output.into_pczt(...).

ConstanceBeguier commented on src/builder.rs:
```diff
@@ -1222,23 +1210,23 @@ impl MaybeSigned {
 }
 
-impl<Proof: fmt::Debug, V, P: OrchardPrimitives> Bundle<InProgress<Proof, Unauthorized>, V, P> {
+impl<P: fmt::Debug, V, Pr: OrchardPrimitives> Bundle<InProgress<P, Unauthorized>, V, Pr> {
```
I would prefer to use the same generic parameter everywhere.
I suggest renaming P to Pr everywhere for OrchardPrimitives.

dmidem replied:
Fixed (renamed `P: OrchardPrimitives` to `Pr: OrchardPrimitives` everywhere in the code).

dmidem committed:
Rename BundleType::DEFAULT_VANILLA back to DEFAULT (r2602373392)

dmidem committed:
Rename P: OrchardPrimitives to Pr everywhere in the code

dmidem committed:
Remove assert_eq from the ActionInfo constructor and return it to build and build_for_pczt methods

dmidem committed:
Rename remaining P: OrchardPrimitives usages to Pr

ConstanceBeguier committed:
Revert "Remove assert_eq from the ActionInfo constructor and return it to build and build_for_pczt methods"

This reverts commit 17de30c5d7a49a2f2bee6aed24bbd99f67fe170d.

ConstanceBeguier committed:
Check spend and output assets are equal in build_for_pczt

ConstanceBeguier committed:
Remove check that spend and output assets are equal in build_for_pczt

Inside ActionInfo::build_for_pczt, we check that
- spend asset is equal to the native asset in self.spend.into_pczt(...)
- output asset is equal to the native asset in
  self.output.into_pczt(...)
